ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -86,6 +90,50 @@ public RemoteReplicaInfo(ReplicaId replicaId, ReplicaId localReplicaId, Store lo
     initializeTokens(token);
     // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
     this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = 0;
+  }
+
+  public RemoteReplicaInfo(ReplicationConfig replicationConfig, ReplicaId replicaId, ReplicaId localReplicaId,
+      Store localStore, FindToken token, long tokenPersistIntervalInMs, Time time, Port port) {
+    this.replicaId = replicaId;
+    this.localReplicaId = localReplicaId;
+    this.totalBytesReadFromLocalStore = 0;
+    this.localStore = localStore;
+    this.time = time;
+    this.port = port;
+    this.tokenPersistIntervalInMs = tokenPersistIntervalInMs;
+    initializeTokens(token);
+    // ExchangeMetadataResponse is initially empty. It will be populated by replica threads during replication cycles.
+    this.exchangeMetadataResponse = new ReplicaThread.ExchangeMetadataResponse(ServerErrorCode.No_Error);
+    this.replicationRetryCount = 0;
+    this.MAX_REPLICATION_RETRY_COUNT = replicationConfig.maxReplicationRetryCount;
+  }
+
+  /**
+   * Increments replicationRetryCount by 1
+   * @return replicationRetryCount
+   */
+  public long incReplicationRetryCount() {
+    this.replicationRetryCount += 1;
```

Comment:
nit: this is redundant.

Reply:
I print the incremented value in a log message, so it helps that this method returns it.

Reply:
I mean `this`, is redundant, you don't have to use `this` in RemoteReplicaInfo class when there is no other variables that have the same name :)

Reply:
understood